### PR TITLE
chore: add clerk example

### PR DESCRIPTION
**Pull request by Pagebakers**
This adds a new example showcasing integrating Clerk with WunderGraph.

This PR depends on #795

### CODE REVIEW

File: docs-website/config/navigation.tsx [hunk 1]
```diff
@@ -90,6 +90,10 @@ const navigation = [
 						title: 'Next.js + React Query',
 						href: '/docs/examples/nextjs-react-query',
 					},
+					{
+						title: 'Next.js + Clerk',
+						href: '/docs/examples/nextjs-clerk',
+					},
```
Comment (jensneuse) [inline docs-website/config/navigation.tsx:96]: Please also add it to the list of examples in the main README.md in the root

File: packages/nextjs/src/auth-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	 * Returning null will unset the token header.
+	 */
+	getToken: () => Promise<string | null | undefined>
+) => MiddlewareReturn;
+
+export const useAuthMiddleware: AuthMiddleware = (useSWRNext, getToken) => {
+	return (key, fetcher, config) => {
+		const context = useWunderGraphContext();
+
+		const fetcherWithAuth: BareFetcher<Promise<unknown>> = async (...args) => {
```
Comment (jensneuse) [inline packages/nextjs/src/auth-middleware.ts:19]: Bare or Bearer?

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: docs-website/src/pages/docs/examples/nextjs-clerk.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+---
+title: Next.js + Clerk Example
+pageTitle: WunderGraph - Examples - Next.js - Clerk
+description: An example that demonstrates how to integrate WunderGraph with Next.js and Clerk
+---
+
+[The NextJS + Clerk example](https://github.com/wundergraph/wundergraph/tree/main/examples/nextjs-clerk) demonstrates how you can use Clerk as your authentication provider with WunderGraph.
+
+This example uses Clerk JWT Templates, Next.js middleware and the WunderGraph Next.js client.
```
Comment (Aenimus) [inline docs-website/src/pages/docs/examples/nextjs-clerk.md:9]: After everything we discussed yesterday :D

File: examples/nextjs-clerk/next-env.d.ts
Comment (Aenimus) [inline examples/nextjs-clerk/next-env.d.ts:1]: does this file serve a purpose?

File: packages/nextjs/src/auth-middleware.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			try {
+				const token = await getToken();
+				if (token) {
+					context?.client.setAuthorizationToken(token);
+				} else if (token === null) {
+					context?.client.unsetAuthorization();
+				}
```
[the remark this answers is not in the document]
Comment (Aenimus) [reply packages/nextjs/src/auth-middleware.ts:26]: Oh, it just stays undefined?

File: packages/nextjs/src/auth-middleware.ts (hunk 2 above)
Comment (Pagebakers) [reply packages/nextjs/src/auth-middleware.ts:19]: Bare is correct

File: examples/nextjs-clerk/next-env.d.ts
Comment (Pagebakers) [reply examples/nextjs-clerk/next-env.d.ts:1]: Yes it part of TypeScript support for nextjs :)

File: packages/nextjs/src/auth-middleware.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Pagebakers) [reply packages/nextjs/src/auth-middleware.ts:26]: Then nothing happens, by design.

File: examples/nextjs-clerk/.wundergraph/wundergraph.operations.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+			...config,
+		}),
+		subscriptions: (config) => ({
+			...config,
+		}),
+		custom: {
+			Country: (config) => ({
+				...config,
+				authentication: {
+					required: true,
```
[the remark this answers is not in the document]
Comment (Pagebakers) [reply examples/nextjs-clerk/.wundergraph/wundergraph.operations.ts:34]: Oh yeah, I copied this straight from the Auth0 example

**Review by Aenimus — APPROVED** [review]
Oxford weeps

**Review by jensneuse — APPROVED** [review]

**Pull request merged**